src/smc-webapp/editor_course.cjsx

```diff
@@ -432,6 +432,12 @@ exports.init_redux = init_redux = (redux, course_project_id, course_filename) ->
             title = "#{store.get_student_name(student_id)} - #{store.getIn(['settings', 'title'])}"
             redux.getActions('projects').set_project_title(student_project_id, title)
 
+        start_all_student_projects: () =>
+            get_store()?.get_students().map (student, student_id) =>
+                student_project_id = student.get('project_id')
+                if student_project_id?
+                    redux.getActions('projects').start_project(student_project_id)
```

Comment:
On Friday, February 19, 2016, Harald Schilly <notifications@github.com>
wrote:

> In src/smc-webapp/editor_course.cjsx
> <https://github.com/sagemathinc/smc/pull/427#discussion_r53445192>:
>
> > @@ -432,6 +432,12 @@ exports.init_redux = init_redux = (redux, course_project_id, course_filename) ->
> >              title = "#{store.get_student_name(student_id)} - #{store.getIn(['settings', 'title'])}"
> >              redux.getActions('projects').set_project_title(student_project_id, title)
> >
> > +        start_all_student_projects: () =>
> > +            get_store()?.get_students().map (student, student_id) =>
> > +                student_project_id = student.get('project_id')
> > +                if student_project_id?
> > +                    redux.getActions('projects').start_project(student_project_id)
>
> I'm just looking very briefly at this. My only concern is if there are
> many student projects, like, more than 100. This map runs sequentially --
> so that could take a lot of time if this start_project call is blocking.
>

Actually that code will do all 100 at once!  It "sequentially" sets the
"start this project" field in the local table.  All at once that is will
get synced to the backend, which will then do all the starts in parallel.
This isn't Python.





> OTOH, it would issue such start commands to all of them at once,
> basically, if it is non-blocking.
>
> Hmm, but I see, it's written in the same way as everywhere else. So, that
> should work then, too …
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sagemathinc/smc/pull/427/files#r53445192>.
>


-- 
Sent from my massive iPhone 6 plus.